arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Platform.kt

```diff
@@ -1,163 +1,18 @@
-      pushAll(stack.iteratorReversed())
-    }
-
-    /** Pops an item from the stack (in LIFO order).
-     *
-     * Returns `null` in case the stack is empty.
-     */
-    fun pop(): A? {
-      if (index == 0) {
-        if (array.getOrNull(0) != null) {
-          array = array[0] as Array<Any?>
-          index = modulo
-        } else {
-          return null
-        }
-      }
-      val result = array[index] as A
-      // GC purposes
-      array[index] = null
-      index -= 1
-      return result
-    }
-
-    /** Builds an iterator out of this stack. */
-    @Suppress("IteratorNotThrowingNoSuchElementException")
-    fun iteratorReversed(): Iterator<A> =
-      object : Iterator<A> {
-        private var array = this@ArrayStack.array
-        private var index = this@ArrayStack.index
-
-        override fun hasNext(): Boolean =
-          index > 0 || (array.getOrNull(0) != null)
-
-        override fun next(): A {
-          if (index == 0) {
-            array = array[0] as Array<Any?>
-            index = modulo
-          }
-          val result = array[index] as A
-          index -= 1
-          return result
-        }
-      }
-
-    fun isNotEmpty(): Boolean =
-      !isEmpty()
-  }
-
-  inline fun <A> onceOnly(crossinline f: (A) -> Unit): (A) -> Unit {
-    val wasCalled = AtomicBooleanW(false)
-
-    return { a ->
-      if (!wasCalled.getAndSet(true)) {
-        f(a)
-      }
-    }
-  }
-
-  internal inline fun <A> onceOnly(conn: SuspendConnection, crossinline f: (A) -> Unit): (A) -> Unit {
-    val wasCalled = AtomicBooleanW(false)
-
-    return { a ->
-      if (!wasCalled.getAndSet(true)) {
-        conn.pop()
-        f(a)
-      }
-    }
-  }
-
-  internal fun <A> unsafeRunSync(f: suspend () -> A): A {
-    val latch = OneShotLatch()
-    var ref: Either<Throwable, A>? = null
-    f.startCoroutine(Continuation(EmptyCoroutineContext) { a ->
-      ref = a.fold({ aa -> Either.Right(aa) }, { t -> Either.Left(t) })
-      latch.releaseShared(1)
-    })
-
-    latch.acquireSharedInterruptibly(1)
-
-    return when (val either = ref) {
-      is Either.Left -> throw either.a
-      is Either.Right -> either.b
-      null -> throw ArrowInternalException("$ArrowExceptionMessage\nSuspend execution should yield a valid result")
-    }
-  }
-
-  internal fun <A> unsafeRunSync(startOn: CoroutineContext, f: suspend () -> A): A {
-    val latch = OneShotLatch()
-    var ref: Either<Throwable, A>? = null
-    f.startCoroutine(Continuation(startOn) { a ->
-      ref = a.fold({ aa -> Either.Right(aa) }, { t -> Either.Left(t) })
-      latch.releaseShared(1)
-    })
-
-    latch.acquireSharedInterruptibly(1)
-
-    return when (val either = ref) {
-      is Either.Left -> throw either.a
-      is Either.Right -> either.b
-      null -> throw ArrowInternalException("$ArrowExceptionMessage\nSuspend execution should yield a valid result")
-    }
-  }
+  internal fun <A> unsafeRunSync(f: suspend () -> A): A =
```

Comment:
Should we also deprecate this and just let people use runBlocking as unsafe method if they wish too? I think most of the users know this by now/

Reply:
Yes, this has always been internal but that's why I `@Deprecated` `Enviroment` type since you should just use structured concurrency to run unsafely side-effects from Java or `runBlocking` to run from `main` or similar.